targets/TARGET_ONSEMI/TARGET_NCS36510/clock.h

```diff
@@ -83,6 +83,7 @@
 #define CLOCK_ENABLE(a)		CLOCKREG->PDIS.WORD &= ~(1 << a)
 #define CLOCK_DISABLE(a)	CLOCKREG->PDIS.WORD |= (uint32_t)(1 << a)
 
+#define CLOCK_IS_ENABLED(a)     ((CLOCKREG->PDIS.WORD >> a) & 1)?0:1
```

Comment:
@0xc0170 suggested in a previous pull request that this implementation be changed from calling an enable function repeatedly. Instead check for the clock and enable only if required. 
I have added enclosing brackets.